# clustalw-portlet

<h2>Requirements</h2>
- JDK 7+
- JSAGA libs (v1.1.x)
- jsaga-job-management (v1.5.96+)

<h2>About the application</h2>
<p align="justify">
<img width="170" src="http://www.abinit.org/logo.png" border="0">
is a widely used multiple program for multiple alignment of nucleic acid and protein sequences. sequence alignment computer program.</br></br>

The program accepts a wide range on input formats including: NBRF/PIR, FASTA, EMBL/Swissprot, Clustal, GCC/MSF, GCG9 RSF, and GDE, and executes the following workflow:</br></br>
- Pairwise alignment
- Creation of a phylogenetic tree (or use a user-defined tree)
- Use of the phylogenetic tree to carry out a multiple alignment
 </br></br>

Users can align the sequences using the default setting but occasionally it may be useful to customize one's own parameters. The main parameters are the gap opening penalty and the gap extension penalty.</br></br>
</p>

<h2>Contribute</h2>
- Fork it
- Create a branch (git checkout -b my_markup)
- Commit your changes (git commit -am "My changes")
- Push to the branch (git push origin my_markup)
- Create an Issue with a link to your branch

<h2>Sponsors</h2>
<p align="justify">
<a href="http://www.infn.it/"><img width="150" src="http://www.infn.it/logo/weblogo1.gif" border="0" title="The INFN Home Page"></a>
</p>
